### PR TITLE
Index select upport 0-Shape

### PR DESCRIPTION
**Pull request by ZzSean**
### PR types

Others
### PR changes

OPs
### Describe

Index select upport 0-Shape.

### CODE REVIEW

File: paddle/phi/kernels/cpu/index_select_kernel.cc [hunk 1]
```diff
@@ -26,6 +26,9 @@ void IndexSelectKernel(const Context& ctx,
                        const DenseTensor& index,
                        int dim,
                        DenseTensor* output) {
+  if (x.numel() == 0 || index.numel() == 0) {
+    return;
```
Comment (FlyingQianMM) [inline paddle/phi/kernels/cpu/index_select_kernel.cc:30]: index_select_grad也需要加下0-shape的支持
if (x.numel() == 0 || out_grad.numel() == 0) {

Comment (ZzSean) [reply paddle/phi/kernels/cpu/index_select_kernel.cc:30]: done, thx

File: paddle/phi/kernels/gpu/index_select_grad_kernel.cu [hunk 2]
```diff
@@ -60,6 +60,9 @@ void IndexSelectGradKernel(const Context& ctx,
                            const DenseTensor& out_grad,
                            int dim,
                            DenseTensor* x_grad) {
+  if (x.numel() == 0 || index.numel() == 0 || out_grad.numel() == 0) {
+    return;
+  }
   auto* output_grad_data = out_grad.data<T>();
   auto* in_grad_data = ctx.template Alloc<T>(x_grad);
```
Comment (FlyingQianMM) [inline paddle/phi/kernels/gpu/index_select_grad_kernel.cu:67]: if (x.numel() == 0 || index.numel() == 0 || out_grad.numel() == 0) {
    return;
  }
要放到这行后面

File: paddle/phi/kernels/gpu/index_select_kernel.cu [hunk 3]
```diff
@@ -49,6 +49,9 @@ void IndexSelectKernel(const Context& ctx,
                        const DenseTensor& index,
                        int dim,
                        DenseTensor* output) {
+  if (x.numel() == 0 || index.numel() == 0) {
+    return;
+  }
```
Comment (FlyingQianMM) [inline paddle/phi/kernels/gpu/index_select_kernel.cu:54]: 需要放到
T* out_data = ctx.template Alloc<T>(output)
后面

**Pull request closed without merging**